.github/workflows/cd.yml

```
@@ -16,113 +16,26 @@ on:
-            type=schedule
-            type=ref,event=branch
-            type=ref,event=pr
-            type=semver,pattern={{version}}
-            type=semver,pattern={{major}}.{{minor}}
-            type=semver,pattern={{major}}
-            type=sha
-
-      # Setup gcloud CLI
-      - name: Authenticate to Google Cloud
-        id: auth
-        uses: google-github-actions/auth@v0.7.1
-        with:
-          workload_identity_provider: 'projects/143793276228/locations/global/workloadIdentityPools/github-actions/providers/github-oidc'
-          service_account: 'github-service-account@zealous-zebra.iam.gserviceaccount.com'
-          token_format: 'access_token'
-
-      - name: Set up QEMU
-        id: qemu
-        uses: docker/setup-qemu-action@v1
-        with:
-          image: tonistiigi/binfmt:latest
-          platforms: all
-
-      # Setup Docker Buildx to allow use of docker cache layers from GH
-      - name: Set up Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v1
-
-      - name: Login to Google Artifact Registry
-        uses: docker/login-action@v1.14.1
-        with:
-          registry: us-docker.pkg.dev
-          username: oauth2accesstoken
-          password: ${{ steps.auth.outputs.access_token }}
-
-      - name: Login to Google Container Registry
-        uses: docker/login-action@v1.14.1
-        with:
-          registry: gcr.io
-          username: oauth2accesstoken
-          password: ${{ steps.auth.outputs.access_token }}
-
-      # Build and push image to Google Artifact Registry
-      - name: Build & push
-        id: docker_build
-        uses: docker/build-push-action@v2.10.0
-        with:
-          target: runtime
-          context: .
-          file: ./docker/Dockerfile
-          # TODO: building crates is taking too long with arm64 and it's timing out on GHA
-          # platforms: |
-          #   linux/amd64
-          #   linux/arm64
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
-          build-args: |
-            NETWORK=${{ github.event.inputs.network || env.NETWORK }}
-            SHORT_SHA=${{ env.GITHUB_SHA_SHORT }}
-            RUST_BACKTRACE=1
-            ZEBRA_SKIP_IPV6_TESTS="1"
-            CHECKPOINT_SYNC=${{ github.event.inputs.checkpoint_sync || true }}
-            SENTRY_DSN=${{ secrets.SENTRY_ENDPOINT }}
-          push: true
-          cache-from: type=registry,ref=${{ env.GAR_BASE }}/${{ env.GITHUB_REF_SLUG_URL }}:buildcache
-          cache-to: type=registry,ref=${{ env.GAR_BASE }}/${{ env.GITHUB_REF_SLUG_URL }}:buildcache,mode=max
+    # TODO: before merging change uses to ZcashFoundation/zebra/.github/workflows/docker-zebra-build.yml@main
+    uses: ./.github/workflows/docker-zebra-build.yml
+    with:
+      # TODO validate if CARGO_INCREMENTAL should be added at build time
+      dockerfile_path: ./docker/Dockerfile
+      dockerfile_target: runtime
+      image_name: main
```

Question: Did this used to be the branch name? What if we're using workflow_dispatch, and the branch we're running this workflow on isn't the `main` branch?

---

Maybe we want something like:
```suggestion
      image_name: zebrad-deploy
```

---

Great question, as I even considered removing the ** workflow_dispatch** from the Continuous deployment workflow.

We don't wan't to trigger deployments to _"Mainnet"_ from a random branch. At worst, because of a deployment failure, we just want to rerun the failed job.

---

The image name has always been `main`, but we can surely change it to `production`, `zebrad-runtime` or `zebrad-deploy` as you suggested.

@dconnolly can help us here for taking the decision.

---

I think this has inherited the branch name, so that things had the branch they built off of (which you can also get for workflow_dispatch)

---

Made `zebrad` the image name